test/Microsoft.ServiceFabric.Diagnostics.Tests/Tracing/UnstructuredTracePublisherTest.cs

```diff
@@ -0,0 +1,157 @@
+            }
+
+            [Theory]
+            [InlineData(ServiceFabricEventSource.ErrorTextEventId, nameof(ITextEventSource.ErrorText), EventLevel.Error)]
+            [InlineData(ServiceFabricEventSource.InfoTextEventId, nameof(ITextEventSource.InfoText), EventLevel.Informational)]
+            [InlineData(ServiceFabricEventSource.NoiseTextEventId, nameof(ITextEventSource.NoiseText), EventLevel.Verbose)]
+            [InlineData(ServiceFabricEventSource.WarningTextEventId, nameof(ITextEventSource.WarningText), EventLevel.Warning)]
+            public void UsesIdAndTypeWhenBothAreIncludedInEventPayload(int eventId, string eventName, EventLevel eventLevel)
+            {
+                EventSource eventSource = CreateTextEventSource();
+                string expectedId = "Id" + fuzzy.String();
+                InsertPayload(0, "id", expectedId);
+                string expectedName = "Type" + fuzzy.String();
+                InsertPayload(1, "type", expectedName);
+                EventWrittenEventArgs eventWritten = CreateEvent(eventSource, eventId, eventName, eventLevel);
+
+                sut.Declared().Method<Action<EventWrittenEventArgs>>().Invoke(eventWritten);
+
+                string expectedTask = eventSource.Name;
+                var expectedLevel = (ushort)eventWritten.Level;
+                var expectedText = string.Format(CultureInfo.InvariantCulture, eventWritten.Message, eventWritten.Payload.ToArray());
+                writeUnstructured.Verify(_ => _.Invoke(expectedTask, expectedName, expectedId, expectedLevel, expectedText));
+            }
+
+            EventWrittenEventArgs CreateEvent(EventSource eventSource)
+            {
+                int id = fuzzy.Int32().Maximum(0); // To prevent EventWrittenEventArgs from fetching EventLevel from uninitialized EventMetadata
+                string name = "Event" + fuzzy.String();
+                var level = fuzzy.Enum<EventLevel>();
+                return CreateEvent(eventSource, id, name, level);
+            }
+
+            EventWrittenEventArgs CreateEvent(EventSource eventSource, int id, string name, EventLevel level)
+            {
+                var @event = Type<EventWrittenEventArgs>.New(eventSource, id);
+                @event.Property<string>(nameof(EventWrittenEventArgs.EventName)).Set(name);
+                @event.Property<EventLevel>().Set(level);
+                @event.Property<ReadOnlyCollection<object>>().Set(new ReadOnlyCollection<object>(payload));
+                @event.Property<ReadOnlyCollection<string>>().Set(new ReadOnlyCollection<string>(payloadNames));
+                @event.Property<string>(nameof(EventWrittenEventArgs.Message)).Set(CreateEventMessage());
+                return @event;
+            }
+
+            string CreateEventMessage()
+            {
+                var message = new StringBuilder("Message " + fuzzy.String().Replace("{", string.Empty).Replace("}", string.Empty));
```

Comment:
I see that `FuzzyString` can accept set of characters. Would it be more readable to create it directly with set of characters we want? Something like `new FuzzyString(fuzzy, length, PrintableAsciiCharacters().Except({ '{', '}' }));`

Reply:
Good point. It's been so long since I've actively coded, I forgot my own API :)